cas-server-support-pac4j/src/main/java/org/apereo/cas/support/pac4j/web/flow/ClientAction.java

```diff
@@ -173,12 +154,16 @@ protected void prepareForLoginPage(final RequestContext context) {
         final Set<ProviderLoginPageConfiguration> urls = new LinkedHashSet<>();
         // for all clients, generate redirection urls
         for (final Client client : this.clients.findAllClients()) {
-            final IndirectClient indirectClient = (IndirectClient) client;
-            // clean Client suffix for default names
-            final String name = client.getName().replace("Client", "");
-            final String redirectionUrl = indirectClient.getRedirectionUrl(webContext);
-            logger.debug("{} -> {}", name, redirectionUrl);
-            urls.add(new ProviderLoginPageConfiguration(name, redirectionUrl, name.toLowerCase()));
+            try {
+                final IndirectClient indirectClient = (IndirectClient) client;
+                // clean Client suffix for default names
+                final String name = client.getName().replace("Client", "");
+                final String redirectionUrl = indirectClient.getRedirectAction(webContext).getLocation();
```

Comment:
I'll raise a warning on this: for OAuth 1.0 and OpenID Connect support, this will require to call the identity server and may generate performance issues.